client/components/recipes/UpdateRecipePage.jsx

```diff
@@ -0,0 +1,327 @@
+ *
+ * @param {Object} event
+ *
+ * @memberof UpdateRecipePage
+ *
+ * @returns {undefined}
+ */
+  onSubmit(event) {
+    event.preventDefault();
+
+    /**
+     * @description Id of the recipe to be updated sent as param
+     */
+    const { id } = this.props.match.params;
+
+    if (this.isValid()) {
+      this.setState({ errors: {} });
+      this.props.updateRecipe(id, this.state)
+        .then(() => {
+          this.setState({ redirect: true });
+        });
+    }
+  }
+
+  /**
+   * @description Handle form validation
+   *
+   * @returns {Object} isValid
+   *
+   * @memberof UpdateRecipePage
+   */
+  isValid() {
+    const { errors, isValid } = validateRecipe(this.state);
+
+    if (!isValid) {
+      this.setState({ errors });
+    }
+
+    return isValid;
+  }
+
+  /**
+ * @description widget function to handle image upload
+ *
+ * @param {Object} event
+ *
+ * @memberof UpdateRecipePage
+ *
+ * @returns {undefined}
+ */
+  uploadWidget(event) {
+    event.preventDefault();
+    const scope = this;
+
+    cloudinary.openUploadWidget(
+      {
+        cloud_name: 'fuchodeveloper',
+        upload_preset: 'wvxnziq0',
+        tags: ['recipe']
+      },
+      (error, result) => {
+        scope.setState({ image: result[0].url });
+      }
+    );
+  }
+
+  /**
+ * @description renders component to DOM
+ *
+ * @memberof UPdateRecipePage
+ *
+ *  @returns {JSX} JSX representation of component
+ */
+  render() {
+    if (this.state.redirect) {
+      const { id } = this.props.match.params;
+      return <Redirect to={`/recipes/${id}`} />;
+    }
+
+    const {
+      name, image, direction, ingredients, isLoading
+    } = this.props.recipe;
+
+    if (isLoading) {
+      return (
+        <h2 className="text-center">Loading...</h2>
+      );
+    }
+
+    return (
+      <div>
+
+        <div className="container margin-top-70">
+          <div>
+            <h1 className="text-center p-4 center-hero-text">
+            Update Recipe
+            </h1>
+          </div>
+
+          <div className="col-md-6 mx-auto p-3">
+            {/* Recipe avatar */}
+            <div className="col-md-6 mx-auto p-3 add_recipe-card text-center">
+              <img
+                src={this.state.image}
+                width="350px"
+                id="recipe-image-avatar"
+                height="auto"
+                className="img-fluid"
+                alt={this.state.name}
+              />
+            </div>
+
+            <form encType="multipart/form-data" onSubmit={this.onSubmit}>
+              <br />
+              <div className="upload text-center">
+                <button
+                  name="image"
+                  id="image"
+                  onClick={this.uploadWidget}
+                  className="upload-button btn-primary btn-primary-color"
+                >
+                      Add Image
+                </button>
+              </div>
+
+              <h3
```

Comment:
If your html tags fit within 80 character limit, keep them on one line and split at the chil-node